diag_manager/fms_diag_yaml_object.F90

```diff
@@ -152,222 +128,206 @@ module fms_diag_yaml_object_mod
- class (diag_yaml_files_var_type), intent(in) :: diag_var_obj !< The object being inquiried
+ class (diagYamlFilesVar_type), intent(in) :: diag_var_obj !< The object being inquiried
  character (len=:), allocatable :: res !< What is returned
   res = diag_var_obj%var_skind
 end function get_var_skind
-!> \brief Inquiry for diag_yaml_files_var_obj%var_outname
+!> @brief Inquiry for diag_yaml_files_var_obj%var_outname
+!! @return var_outname of a diag_yaml_files_var_obj
 pure function get_var_outname (diag_var_obj) result (res)
- class (diag_yaml_files_var_type), intent(in) :: diag_var_obj !< The object being inquiried
+ class (diagYamlFilesVar_type), intent(in) :: diag_var_obj !< The object being inquiried
  character (len=:), allocatable :: res !< What is returned
   res = diag_var_obj%var_outname
 end function get_var_outname
-!> \brief Inquiry for diag_yaml_files_var_obj%var_longname
+!> @brief Inquiry for diag_yaml_files_var_obj%var_longname
+!! @return var_longname of a diag_yaml_files_var_obj
 pure function get_var_longname (diag_var_obj) result (res)
- class (diag_yaml_files_var_type), intent(in) :: diag_var_obj !< The object being inquiried
+ class (diagYamlFilesVar_type), intent(in) :: diag_var_obj !< The object being inquiried
  character (len=:), allocatable :: res !< What is returned
   res = diag_var_obj%var_longname
 end function get_var_longname
-!> \brief Inquiry for diag_yaml_files_var_obj%var_units
+!> @brief Inquiry for diag_yaml_files_var_obj%var_units
+!! @return var_units of a diag_yaml_files_var_obj
 pure function get_var_units (diag_var_obj) result (res)
- class (diag_yaml_files_var_type), intent(in) :: diag_var_obj !< The object being inquiried
+ class (diagYamlFilesVar_type), intent(in) :: diag_var_obj !< The object being inquiried
  character (len=:), allocatable :: res !< What is returned
   res = diag_var_obj%var_units
 end function get_var_units
-!> \brief Inquiry for diag_yaml_files_var_obj%var_write
+!> @brief Inquiry for diag_yaml_files_var_obj%var_write
+!! @return var_write of a diag_yaml_files_var_obj
 pure function get_var_write (diag_var_obj) result (res)
- class (diag_yaml_files_var_type), intent(in) :: diag_var_obj !< The object being inquiried
+ class (diagYamlFilesVar_type), intent(in) :: diag_var_obj !< The object being inquiried
  logical :: res !< What is returned
   res = diag_var_obj%var_write
 end function get_var_write
-!> \brief Inquiry for diag_yaml_files_var_obj%var_attributes
+!> @brief Inquiry for diag_yaml_files_var_obj%var_attributes
+!! @return var_attributes of a diag_yaml_files_var_obj
 pure function get_var_attributes(diag_var_obj) result (res)
- class (diag_yaml_files_var_type), intent(in) :: diag_var_obj !< The object being inquiried
- character (len=:), allocatable :: res (:) !< What is returned
+ class (diagYamlFilesVar_type), intent(in) :: diag_var_obj !< The object being inquiried
+ character (len=MAX_STR_LEN), allocatable :: res (:,:) !< What is returned
```

Comment:
I think you should add a comment about why len=MAX_STR_LEN and not : for someone else who will have to deal with this in the future.